mmv1/third_party/terraform/services/dataproc/resource_dataproc_cluster.go

```diff
@@ -1139,6 +1139,74 @@ func ResourceDataprocCluster() *schema.Resource {
 										Elem:        &schema.Schema{Type: schema.TypeString},
 										Description: `List of preemptible instance names which have been assigned to the cluster.`,
 									},
+									"instance_flexibility_policy": {
+										Type:        schema.TypeList,
+										Optional:    true,
+										Computed:    true,
+										Description: `Instance flexibility Policy allowing a mixture of VM shapes and provisioning models.`,
+										AtLeastOneOf: []string{
+											"cluster_config.0.preemptible_worker_config.0.num_instances",
+											"cluster_config.0.preemptible_worker_config.0.preemptibility",
+											"cluster_config.0.preemptible_worker_config.0.disk_config",
+											"cluster_config.0.preemptible_worker_config.0.instance_flexibility_policy",
+										},
+										MaxItems: 1,
+										Elem: &schema.Resource{
+											Schema: map[string]*schema.Schema{
+												"instance_selection_list": {
+													Type:     schema.TypeList,
+													Computed: true,
+													Optional: true,
+													ForceNew: true,
+													AtLeastOneOf: []string{
```

Comment:
AtLeastOneOf usually requires multiple entries. If this is the only entry, should it just be required?

Reply:
Hi Cameron, thanks for the quick update.
The reason I made it single entry instead of "Required: true" is that there are other configurations WIP under instance_flexibility_policy, which will be added into the AtLeastOneOf list after GA.

Reply:
That's fine then.